addons/mail/models/mail_thread.py

```diff
@@ -1805,25 +1805,33 @@ def _message_add_suggested_recipient(self, result, partner=None, email=None, lan
                 reason, create_values """
         self.ensure_one()
         partner_info = {}
+        recipient_data = {'lang': lang, 'reason': reason}
         if email and not partner:
             # get partner info from email
             partner_info = self._message_partner_info_from_emails([email])[0]
             if partner_info.get('partner_id'):
                 partner = self.env['res.partner'].sudo().browse([partner_info['partner_id']])[0]
-        if email and email in [val[1] for val in result[self.ids[0]]]:  # already existing email -> skip
+        if email and email in [val['email'] for val in result[self.ids[0]]]:  # already existing email -> skip
             return result
         if partner and partner in self.message_partner_ids:  # recipient already in the followers -> skip
             return result
-        if partner and partner.id in [val[0] for val in result[self.ids[0]]]:  # already existing partner ID -> skip
+        if partner and partner.id in [val['persona'].get('id') for val in result[self.ids[0]] if
+                                      val.get('persona', {}).get('type') == 'partner']:  # already existing partner ID -> skip
             return result
         if partner and partner.email:  # complete profile: id, name <email>
-            result[self.ids[0]].append((partner.id, partner.email_formatted, lang, reason, {}))
+            email_normalized = ','.join(email_normalize_all(partner.email))
+            recipient_data.update(
+                {'name': partner.name, 'email': email_normalized, 'persona': {'type': 'partner', 'id': partner.id}})
         elif partner:  # incomplete profile: id, name
-            result[self.ids[0]].append((partner.id, partner.name or '', lang, reason, {}))
+            recipient_data.update(
+                {'name': partner.name, 'email': None, 'persona': {'type': 'partner', 'id': partner.id}})
         else:  # unknown partner, we are probably managing an email address
             _, parsed_email_normalized = parse_contact_from_email(email)
             partner_create_values = self._get_customer_information().get(parsed_email_normalized, {})
-            result[self.ids[0]].append((False, partner_info.get('full_name') or email, lang, reason, partner_create_values))
+            name = partner_create_values.get('name') or partner_info.get('full_name') or email
+            recipient_data.update({'name': name, 'email': partner_info.get('full_name') or email,
+                                   'defaultCreateValues': partner_create_values})
```

Comment:
Use python case + simplify naming
```suggestion
                                   'create_values': partner_create_values})
```